diff --git a/includes/Job/MassMessageJob.php b/includes/Job/MassMessageJob.php
index 634c194..8e52db4 100644
--- a/includes/Job/MassMessageJob.php
+++ b/includes/Job/MassMessageJob.php
@@ -316,6 +316,13 @@ class MassMessageJob extends Job {
 			&& defined( 'NS_TOPIC' )
 			&& !$targetPage->inNamespace( NS_TOPIC );
 
+		// Allow extensions to execute code before this message is sent, and potentially implement
+		// their own messaging system.
+		// Also, if the page is using a different discussion system, handle it specially.
+		if (!Hooks::run('MassMessageJobBeforeMessageSent', array($this))) {
+			return true;
+		}
+
 		// If the page is using a different discussion system, handle it specially
 		if ( $isLqtThreads || $isStructuredDiscussion ) {
 			$subject = $messageBuilder->buildPlaintextSubject( $subject, $pageSubject );
